allensdk/brain_observatory/behavior/behavior_project_cache.py

```diff
@@ -0,0 +1,311 @@
+                lambda x: np.fromstring(x, sep="|", dtype=int))
+            return df
+
+        sessions = self._get_session_summary(
+            self.fetch_api.get_session_table, self.OPHYS_SESSIONS_KEY,
+            write_csv_, read_csv_)
+        sessions = sessions.rename(columns={"genotype": "full_genotype"})
+        if suppress:
+            sessions.drop(columns=suppress, inplace=True, errors="ignore")
+        return sessions
+
+    def add_manifest_paths(self, manifest_builder):
+        manifest_builder = super().add_manifest_paths(manifest_builder)
+        for key, config in self.MANIFEST_CONFIG.items():
+            manifest_builder.add_path(key, **config)
+        return manifest_builder
+
+    def get_behavior_session_table(
+            self,
+            suppress: Optional[List[str]] = None) -> pd.DataFrame:
+        """
+        Return summary table of all behavior_session_ids in the database.
+        :param suppress: optional list of columns to drop from the resulting
+            dataframe.
+        :type suppress: list of str
+        :rtype: pd.DataFrame
+        """
+        read_csv_ = partial(read_csv, index_col="behavior_session_id")
+        sessions = self._get_session_summary(
+            self.fetch_api.get_behavior_only_session_table,
+            self.BEHAVIOR_SESSIONS_KEY, write_csv, read_csv_)
+        sessions = sessions.rename(columns={"genotype": "full_genotype"})
+        if suppress:
+            sessions.drop(columns=suppress, inplace=True, errors="ignore")
+        return sessions
+
+    def get_ophys_experiment_table(
+            self,
+            suppress: Optional[List[str]] = None) -> pd.DataFrame:
+        """
+        Returns the same summary table as `get_session_table`,
+        but with values for "ophys_experiment_id" exploded. Indexed to
+        "ophys_experiment_id."
+        :param suppress: optional list of columns to drop from the resulting
+            dataframe.
+        :type suppress: list of str
+        :rtype: pd.DataFrame
+        """
+        sessions = self.get_session_table(suppress)
+        experiments = (sessions.reset_index()
+                       .explode("ophys_experiment_id")
+                       .set_index("ophys_experiment_id"))
+        return experiments
+
+    def _get_session_summary(
+            self,
+            fetch_call: Callable[[], pd.DataFrame],
+            cache_key: str,
+            write_csv: Callable[[str], None],
+            read_csv: Callable[[str], pd.DataFrame]) -> pd.DataFrame:
+        """
+        Generic helper method for caching calls to get session summary data,
+        such as `get_behavior_session_table` and `get_session_table`.
+        """
+        path = self.get_cache_path(None, cache_key)
+        response = one_file_call_caching(path, fetch_call, write_csv, read_csv)
+        return response
+
+    def get_session_data(self, ophys_experiment_id: int, fixed: bool = False):
```

Comment:
might want to add a warning if fixed is True pointing out that the data itself might not be fixed.